### ISSUE TITLE
Grammar issue: If a query uses UNION, the ORDER BY should apply to the whole result set, not the last query

### ISSUE THREAD

**Issue by SuperBo** [sqlfluff/sqlfluff#852]
`
SELECT City FROM Customers
UNION
SELECT City FROM Suppliers
ORDER BY City
`

## Expected Behaviour
*orderby_clause* should be in the right under *set_expression*.

## Observed Behaviour
*orderby_clause* is two level under *set_expression*, under the last *select_statement*

the output should be the same as following query as mentioned here https://dev.mysql.com/doc/refman/8.0/en/union.html

`
(SELECT City FROM Customers)
UNION
(SELECT City FROM Suppliers)
ORDER BY City
`

## Steps to Reproduce
```python
query = """
SELECT City FROM Customers
UNION
SELECT City FROM Suppliers
ORDER BY City
"""
r = Linter(dialect='mysql').parse_string(query)
r.tree.to_tuple()
```

## Version
Include the output of `sqlfluff --version` along with your Python version
Python: 3.9.1
sqlfluff: 0.4.1

**Comment by barrywhart** [comment]
In other words, with a `UNION` query, the `ORDER BY` applies to the entire result set, not the last query.

https://stackoverflow.com/questions/4715820/how-to-order-by-with-union-in-sql

@SuperBo: I agree that the query is being parsed incorrectly. Can you explain how this is affecting your use of SQLFluff? That will help us to determine the priority of this issue.

**Comment by alanmcruickshank** [comment]
Agreed that we should change the parse structure to default to the order by and limit clauses in the outer structure.

Snowflake agrees: https://docs.snowflake.com/en/sql-reference/operators-query.html

This also makes me wonder whether:
- if an order by and limit clause are provided in the first of the two queries, are they just ignored or does it raise a syntax error?
- if any ORDER BY within a set operation is legal but effectively pointless, should we have a lighting rule which flags this as a violation? (Unless also using a limit clause, in which case I could see the value)

**Comment by barrywhart** [comment]
Re:
>if an order by and limit clause are provided in the first of the two queries, are they just ignored or does it raise a syntax error?

I tested this on BigQuery and got a syntax error "Expected end of input".
```
SELECT 1 AS a
ORDER BY a
UNION ALL
SELECT 1 AS a
ORDER BY a
```

This query also reports a syntax error, in spite of the `LIMIT` clause (which surprises me).
```
SELECT 
job_id
FROM `rsg-engagement-metrics-prod.clv_reports.store_report`
ORDER BY job_id
LIMIT 1
UNION ALL
SELECT 1 AS a
ORDER BY a
```

Adding parentheses around the first query does *not* report a syntax error:
```
(SELECT 
job_id
FROM `rsg-engagement-metrics-prod.clv_reports.store_report`
ORDER BY job_id
LIMIT 1)
UNION ALL
SELECT 1 AS a
ORDER BY a
```

**Comment by SuperBo** [comment]
Dear @barrywhart : I am building a SQL AST parser that based on sqlfluff parsed tree. Then if the parse tree is incorrect, my AST will also be.

**Comment by barrywhart** [comment]
@alanmcruickshank: Do you have thoughts on how to implement this? I'm not that familiar with how to force precedence with parsing rules. Is this potentially similar to the recent work to change the parsing structure of `JOIN` clauses.

**Comment by alanmcruickshank** [comment]
I'll pick this one up. I've got a few ideas.
